src/Control/Selective/Trans/Except.hs

```diff
@@ -0,0 +1,110 @@
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE PackageImports #-}
+{- | A newtype around @transformers@ 'ExceptT' with less restrictive 'Applicative', 'Selective', and 'Alternative' implementations.
+
+Supplies an @instance 'Selective' f => 'Selective' ('ExceptT' e f)@.
+In other words, 'ExceptT' is a bona-fide 'Selective' transformer.
+
+This tries to copy verbatim the API from @transformers@,
+so it can be used as a drop-in replacement.
+The documentation can be found in the [@transformers@](https://hackage.haskell.org/package/transformers/docs/Control-Monad-Trans-Except.html) package.
+-}
+module Control.Selective.Trans.Except where
+
+import Control.Applicative (Alternative (empty, (<|>)))
+import Control.Monad (MonadPlus)
+import Control.Monad.Fix (MonadFix)
+import Control.Monad.IO.Class (MonadIO)
+import Control.Monad.Zip (MonadZip)
+import Data.Functor.Classes
+import Data.Functor.Contravariant (Contravariant)
+import Data.Functor.Identity
+
+import qualified "transformers" Control.Monad.Trans.Except as Transformers
+import Control.Monad.Trans.Class
+
+import Control.Selective
+import Control.Monad.Signatures
+
+-- | A newtype around @transformers@' 'Transformers.ExceptT'.
+newtype ExceptT e m a = ExceptT
+    { toTransformers :: Transformers.ExceptT e m a }
+    deriving (Functor, Monad, MonadTrans, MonadFix, MonadFail, Foldable, Contravariant, Eq1, Ord1, Read1, Show1, MonadZip, MonadIO, MonadPlus, Eq, Ord, Read, Show)
+
+instance Traversable f => Traversable (ExceptT e f) where
+    traverse f (ExceptT efa)= ExceptT <$> traverse f efa
+
+-- | Convert back to the newtype.
+fromTransformers :: Transformers.ExceptT e m a -> ExceptT e m a
+fromTransformers = ExceptT
+
+-- TODO derive instances via transformers
+
+-- | No @'Monad' f@ constraint is needed.
+instance Selective f => Applicative (ExceptT e f) where
+    pure = ExceptT . Transformers.ExceptT . pure . pure
+    ExceptT (Transformers.ExceptT f) <*> ExceptT (Transformers.ExceptT m) = ExceptT $ Transformers.ExceptT
+      $ either (Right . Left) Left <$> f
+      <*? (flip fmap <$> m)
+
+-- | No @'Monad' f@ constraint is needed.
+instance Selective f => Selective (ExceptT e f) where
+    select (ExceptT (Transformers.ExceptT meab)) (ExceptT (Transformers.ExceptT mef)) = ExceptT $ Transformers.ExceptT
+        $ commute <$> meab
+        <*? (swapFunctionEither <$> mef)
+        where
+            commute :: Either e (Either a b) -> Either a (Either e b)
+            commute (Left e) = Right (Left e)
+            commute (Right (Left a)) = Left a
+            commute (Right (Right b)) = Right (Right b)
+
+            swapFunctionEither :: Either e (a -> b) -> a -> Either e b
```

Comment:
I can't quite figure out why you called this function this way. Perhaps something like `applyRight` would be clearer.

Reply:
Because I'm swapping the order of `->` and `Either` :sweat_smile:. Maybe `commuteEitherAndArrow` would have been clearer. But `applyRight` is also good.

Reply:
Ah, got it :) Feel free to use the name you like -- I might just make some renamings in future.